tests/epyccel/test_array_as_func_args.py

```diff
@@ -0,0 +1,144 @@
+"""
+This test file is made for testing the functionality of passing numpy arrays
+in the function arguments.
+"""
+
+import pytest
+import numpy as np
+from numpy.random import randint, uniform
+
+from pyccel.epyccel import epyccel
+from pyccel.decorators import types
+
+int_types = ['int8', 'int16', 'int32', 'int64']
+float_types = ['float32', 'float64']
+
+def test_array_int_1d_scalar_add(language):
+    @types( 'int8[:]' , 'int8' , 'int')
+    @types( 'int16[:]', 'int16', 'int')
+    @types( 'int32[:]', 'int32', 'int')
+    @types( 'int64[:]', 'int64', 'int')
+    def array_int_1d_scalar_add( x, a, x_len ):
+        for i in range(x_len):
+            x[i] += a
+    f1 = array_int_1d_scalar_add
+    f2 = epyccel(f1, language=language)
+
+    for t in int_types:
+        size = randint(1, 30)
+        x1 = randint(np.iinfo(t).max / 2, size=size, dtype=t)
+        x2 = np.copy(x1)
+        a = randint(np.iinfo(t).max / 2, dtype=t)
+
+        f1(x1, a, size)
+        f2(x2, a, size)
+
+    assert np.array_equal( x1, x2 )
```

Comment:
Shouldn't this assertion be in the for loop?

Reply:
yap, I missed it.